### DISCUSSION TITLE
arm64 - MESA: error: ZINK: vkCreateInstance failed (VK_ERROR_INCOMPATIBLE_DRIVER) ib-gateway-1  | glx: failed to create drisw screen

### DISCUSSION THREAD

**Discussion by nadomani** [gnzsnz/ib-gateway-docker#193] [Q&A]
Added platform: linux/amd64 but seems like there is an error in the building process with waiting for config dialog future to complete:
ateway-1  | Have you tried the x11vnc '-ncache' VNC client-side pixel caching feature yet?
ib-gateway-1  | 
ib-gateway-1  | The scheme stores pixel data offscreen on the VNC viewer side for faster
ib-gateway-1  | retrieval.  It should work with any VNC viewer.  Try it by running:
ib-gateway-1  | 
ib-gateway-1  |     x11vnc -ncache 10 ...
ib-gateway-1  | 
ib-gateway-1  | One can also add -ncache_cr for smooth 'copyrect' window motion.
ib-gateway-1  | More info: http://www.karlrunge.com/x11vnc/faq.html#faq-client-caching
ib-gateway-1  | 
ib-gateway-1  | IBC Settings:
ib-gateway-1  |     AcceptBidAskLastSizeDisplayUpdateNotification=accept
ib-gateway-1  |     AcceptIncomingConnectionAction=manual
ib-gateway-1  |     AcceptNonBrokerageAccountWarning=yes
ib-gateway-1  |     AllowBlindTrading=no
ib-gateway-1  |     AutoLogoffTime=
ib-gateway-1  |     AutoRestartTime=11:59 PM
ib-gateway-1  |     BindAddress=
ib-gateway-1  |     BypassBondWarning=
ib-gateway-1  |     BypassCalledBondWarning=
ib-gateway-1  |     BypassNegativeYieldToWorstConfirmation=
ib-gateway-1  |     BypassNoOverfillProtectionPrecaution=
ib-gateway-1  |     BypassOrderPrecautions=
ib-gateway-1  |     BypassPriceBasedVolatilityRiskWarning=
ib-gateway-1  |     BypassRedirectOrderWarning=
ib-gateway-1  |     BypassSameActionPairTradeWarning=
ib-gateway-1  |     BypassUSStocksMarketDataInSharesWarning=
ib-gateway-1  |     ClosedownAt=
ib-gateway-1  |     ColdRestartTime=
ib-gateway-1  |     CommandPrompt=
ib-gateway-1  |     CommandServerPort=0
ib-gateway-1  |     ConfirmCryptoCurrencyOrders=transmit
ib-gateway-1  |     ConfirmOrderIdReset=
ib-gateway-1  |     ControlFrom=
ib-gateway-1  |     DismissNSEComplianceNotice=yes
ib-gateway-1  |     DismissPasswordExpiryWarning=no
ib-gateway-1  |     ExistingSessionDetectedAction=primary
ib-gateway-1  |     ExitAfterSecondFactorAuthenticationTimeout=no
ib-gateway-1  |     FIX=no
ib-gateway-1  |     FIXLoginId=***
ib-gateway-1  |     FIXPassword=***
ib-gateway-1  |     IbDir=
ib-gateway-1  |     IbLoginId=***
ib-gateway-1  |     IbPassword=***
ib-gateway-1  |     LogStructureScope=known
ib-gateway-1  |     LogStructureWhen=never
ib-gateway-1  |     LoginDialogDisplayTimeout=60
ib-gateway-1  |     MinimizeMainWindow=no
ib-gateway-1  |     OverrideTwsApiPort=
ib-gateway-1  |     OverrideTwsMasterClientID=
ib-gateway-1  |     ReadOnlyApi=no
ib-gateway-1  |     ReadOnlyLogin=no
ib-gateway-1  |     ReloginAfterSecondFactorAuthenticationTimeout=yes
ib-gateway-1  |     ResetOrderIdsAtStart=
ib-gateway-1  |     SaveTwsSettingsAt=
ib-gateway-1  |     SecondFactorAuthenticationExitInterval=60
ib-gateway-1  |     SecondFactorAuthenticationTimeout=180
ib-gateway-1  |     SecondFactorDevice=
ib-gateway-1  |     SendMarketDataInLotsForUSstocks=
ib-gateway-1  |     StoreSettingsOnServer=
ib-gateway-1  |     SuppressInfoMessages=yes
ib-gateway-1  |     TradingMode=paper
ib-gateway-1  |     TrustedTwsApiClientIPs=
ib-gateway-1  | End IBC Settings
ib-gateway-1  | 
ib-gateway-1  | 2024-10-26 22:37:00:182 IBC: version: 3.20.0
ib-gateway-1  | System Properties
ib-gateway-1  | ------------------------------------------------------------
ib-gateway-1  | jtsConfigDir = /home/ibgateway/Jts
ib-gateway-1  | java.runtime.name = Java(TM) SE Runtime Environment
ib-gateway-1  | sun.boot.library.path = /usr/local/i4j_jres/Oda-jK0QgTEmVssfllLP/1.8.0_202/lib/amd64
ib-gateway-1  | java.vm.version = 25.202-b08
ib-gateway-1  | java.vm.vendor = Oracle Corporation
ib-gateway-1  | java.vendor.url = http://java.oracle.com/
ib-gateway-1  | path.separator = :
ib-gateway-1  | java.vm.name = Java HotSpot(TM) 64-Bit Server VM
ib-gateway-1  | file.encoding.pkg = sun.io
ib-gateway-1  | user.country = US
ib-gateway-1  | sun.java.launcher = SUN_STANDARD
ib-gateway-1  | sun.os.patch.level = unknown
ib-gateway-1  | java.vm.specification.name = Java Virtual Machine Specification
ib-gateway-1  | user.dir = /home/ibgateway/Jts
ib-gateway-1  | java.runtime.version = 1.8.0_202-b08
ib-gateway-1  | ibcsessionid = AZP7HJJv
ib-gateway-1  | java.awt.graphicsenv = sun.awt.X11GraphicsEnvironment
ib-gateway-1  | java.endorsed.dirs = /usr/local/i4j_jres/Oda-jK0QgTEmVssfllLP/1.8.0_202/lib/endorsed
ib-gateway-1  | os.arch = amd64
ib-gateway-1  | java.io.tmpdir = /tmp
ib-gateway-1  | line.separator = 
ib-gateway-1  | 
ib-gateway-1  | java.vm.specification.vendor = Oracle Corporation
ib-gateway-1  | os.name = Linux
ib-gateway-1  | sun.jnu.encoding = ANSI_X3.4-1968
ib-gateway-1  | java.library.path = /usr/java/packages/lib/amd64:/usr/lib64:/lib64:/lib:/usr/lib
ib-gateway-1  | java.specification.name = Java Platform API Specification
ib-gateway-1  | java.class.version = 52.0
ib-gateway-1  | sun.management.compiler = HotSpot 64-Bit Tiered Compilers
ib-gateway-1  | exe4j.isInstall4j = true
ib-gateway-1  | os.version = 6.10.4-linuxkit
ib-gateway-1  | user.home = /home/ibgateway
ib-gateway-1  | user.timezone = Europe/Zurich
ib-gateway-1  | java.awt.printerjob = sun.print.PSPrinterJob
ib-gateway-1  | file.encoding = ANSI_X3.4-1968
ib-gateway-1  | java.specification.version = 1.8
ib-gateway-1  | java.class.path = /home/ibgateway/Jts/ibgateway/10.32.1i/jars/batik-all-1.16.jar:/home/ibgateway/Jts/ibgateway/10.32.1i/jars/commons-io-2.11.0.jar:/home/ibgateway/Jts/ibgateway/10.32.1i/jars/fop-2.8.jar:/home/ibgateway/Jts/ibgateway/10.32.1i/jars/jackson-annotations-2.12.3.jar:/home/ibgateway/Jts/ibgateway/10.32.1i/jars/jackson-core-2.12.3.jar:/home/ibgateway/Jts/ibgateway/10.32.1i/jars/jackson-databind-2.12.3.jar:/home/ibgateway/Jts/ibgateway/10.32.1i/jars/jts4launch-1032.jar:/home/ibgateway/Jts/ibgateway/10.32.1i/jars/jxbrowser-7.29.jar:/home/ibgateway/Jts/ibgateway/10.32.1i/jars/jxbrowser-linux64-7.29.jar:/home/ibgateway/Jts/ibgateway/10.32.1i/jars/jxbrowser-swing-7.29.jar:/home/ibgateway/Jts/ibgateway/10.32.1i/jars/locales.jar:/home/ibgateway/Jts/ibgateway/10.32.1i/jars/log4j-api-2.17.1.jar:/home/ibgateway/Jts/ibgateway/10.32.1i/jars/log4j-core-2.17.1.jar:/home/ibgateway/Jts/ibgateway/10.32.1i/jars/total-2023.jar:/home/ibgateway/Jts/ibgateway/10.32.1i/jars/twslaunch-1032.jar:/home/ibgateway/Jts/ibgateway/10.32.1i/jars/twslaunch-install4j-1.12.jar:/home/ibgateway/Jts/ibgateway/10.32.1i/jars/xmlgraphics-commons-2.8.jar:/home/ibgateway/Jts/ibgateway/10.32.1i/.install4j/i4jruntime.jar:/home/ibgateway/ibc/IBC.jar
ib-gateway-1  | user.name = ibgateway
ib-gateway-1  | channel = latest
ib-gateway-1  | java.vm.specification.version = 1.8
ib-gateway-1  | sun.java.command = ibcalpha.ibc.IbcGateway /home/ibgateway/ibc/config.ini
ib-gateway-1  | java.home = /usr/local/i4j_jres/Oda-jK0QgTEmVssfllLP/1.8.0_202
ib-gateway-1  | sun.arch.data.model = 64
ib-gateway-1  | user.language = en
ib-gateway-1  | java.specification.vendor = Oracle Corporation
ib-gateway-1  | awt.toolkit = sun.awt.X11.XToolkit
ib-gateway-1  | java.vm.info = mixed mode
ib-gateway-1  | java.version = 1.8.0_202
ib-gateway-1  | java.ext.dirs = /usr/local/i4j_jres/Oda-jK0QgTEmVssfllLP/1.8.0_202/lib/ext:/usr/java/packages/lib/ext
ib-gateway-1  | sun.boot.class.path = /usr/local/i4j_jres/Oda-jK0QgTEmVssfllLP/1.8.0_202/lib/resources.jar:/usr/local/i4j_jres/Oda-jK0QgTEmVssfllLP/1.8.0_202/lib/rt.jar:/usr/local/i4j_jres/Oda-jK0QgTEmVssfllLP/1.8.0_202/lib/sunrsasign.jar:/usr/local/i4j_jres/Oda-jK0QgTEmVssfllLP/1.8.0_202/lib/jsse.jar:/usr/local/i4j_jres/Oda-jK0QgTEmVssfllLP/1.8.0_202/lib/jce.jar:/usr/local/i4j_jres/Oda-jK0QgTEmVssfllLP/1.8.0_202/lib/charsets.jar:/usr/local/i4j_jres/Oda-jK0QgTEmVssfllLP/1.8.0_202/lib/jfr.jar:/usr/local/i4j_jres/Oda-jK0QgTEmVssfllLP/1.8.0_202/classes
ib-gateway-1  | java.vendor = Oracle Corporation
ib-gateway-1  | file.separator = /
ib-gateway-1  | twslaunch.autoupdate.serviceImpl = com.ib.tws.twslaunch.install4j.Install4jAutoUpdateService
ib-gateway-1  | java.vendor.url.bug = http://bugreport.sun.com/bugreport/
ib-gateway-1  | install4jType = standalone
ib-gateway-1  | sun.io.unicode.encoding = UnicodeLittle
ib-gateway-1  | sun.cpu.endian = little
ib-gateway-1  | sun.cpu.isalist = 
ib-gateway-1  | ------------------------------------------------------------
ib-gateway-1  | 2024-10-26 22:37:00:217 IBC: using default settings provider: ini file is /home/ibgateway/ibc/config.ini
ib-gateway-1  | 2024-10-26 22:37:00:217 IBC: using default login manager: will get username and password from settings; FIX username and password (if required) from settings
ib-gateway-1  | 2024-10-26 22:37:00:218 IBC: using default main window manager
ib-gateway-1  | 2024-10-26 22:37:00:218 IBC: using default trading mode manager: constructor parameter args but trading mode not present - will be taken from settings
ib-gateway-1  | 2024-10-26 22:37:00:223 IBC: using default config dialog manager
ib-gateway-1  | 2024-10-26 22:37:00:241 IBC: CommandServer is not started because the port is not configured
ib-gateway-1  | 2024-10-26 22:37:00:561 IBC: TWS Settings directory is: /home/ibgateway/Jts
ib-gateway-1  | 2024-10-26 22:37:00:562 IBC: Starting session: will exit if login dialog is not displayed within 60 seconds
ib-gateway-1  | 2024-10-26 22:37:00:607 IBC: Ensuring /home/ibgateway/Jts/jts.ini contains required minimal lines
ib-gateway-1  | 2024-10-26 22:37:00:608 IBC: Found setting: [Logon]/s3store=true
ib-gateway-1  | 2024-10-26 22:37:00:608 IBC: Found setting: [Logon]/Locale=en
ib-gateway-1  | 2024-10-26 22:37:00:608 IBC: Found setting: [Logon]/displayedproxymsg=1
ib-gateway-1  | 2024-10-26 22:37:00:609 IBC: Found setting: [Logon]/UseSSL=true
ib-gateway-1  | 2024-10-26 22:37:00:609 IBC: Found setting: [IBGateway]/ApiOnly=true
ib-gateway-1  | 2024-10-26 22:37:00:610 IBC: Found setting: [IBGateway]/TrustedIPs=127.0.0.1
ib-gateway-1  | 2024-10-26 22:37:00:610 IBC: Confirmed /home/ibgateway/Jts/jts.ini contains required minimal lines
ib-gateway-1  | 2024-10-26 22:37:00:611 IBC: Starting Gateway
ib-gateway-1  | MESA: error: ZINK: vkCreateInstance failed (VK_ERROR_INCOMPATIBLE_DRIVER)
ib-gateway-1  | glx: failed to create drisw screen
ib-gateway-1  | 2024-10-26 22:37:08:699 IBC: Getting config dialog
ib-gateway-1  | 2024-10-26 22:37:08:702 IBC: Creating config dialog future
ib-gateway-1  | 2024-10-26 22:37:08:713 IBC: Getting config dialog
ib-gateway-1  | 2024-10-26 22:37:08:715 IBC: Getting main window
ib-gateway-1  | 2024-10-26 22:37:08:716 IBC: Creating main window future
ib-gateway-1  | 2024-10-26 22:37:08:716 IBC: Waiting for config dialog future to complete

**Comment by gnzsnz** [comment]
Hi,

not sure what you are trying to do or what you expect. by doing some guess work i believe that you have aarch64/arm and want to get this image running. if that's the case setting architecture to amd64 won't work.

there is and **EXPERIMENTAL** branch for aarch64 #102  but it has many problems. You can give it a try and work your way through the problems.

if you find a way to have it working please share what you did.

some useful links maybe --> #81 ,#134 , #113

**Reply by nadomani** [reply]
So currently there is no option to run the docker image on a Mac with apple chip? (M2)

**Reply by gnzsnz** [reply]
yes there are, they just a bit involved. #102  is one, and there is this option using emulation https://github.com/gnzsnz/ib-gateway-docker/issues/81#issuecomment-2440064087
